aiosmtpd/smtp.py

```diff
@@ -352,28 +352,33 @@ def _set_rset_state(self):
         if not hostname:
             await self.push('501 Syntax: EHLO hostname')
             return
+
+        response = []
         self._set_rset_state()
         self.session.extended_smtp = True
-        await self.push('250-%s' % self.hostname)
+        response.append('250-%s' % self.hostname)
         if self.data_size_limit:
-            await self.push('250-SIZE %s' % self.data_size_limit)
+            response.append('250-SIZE %s' % self.data_size_limit)
             self.command_size_limits['MAIL'] += 26
         if not self._decode_data:
-            await self.push('250-8BITMIME')
+            response.append('250-8BITMIME')
         if self.enable_SMTPUTF8:
-            await self.push('250-SMTPUTF8')
+            response.append('250-SMTPUTF8')
             self.command_size_limits['MAIL'] += 10
         if self.tls_context and not self._tls_protocol:
-            await self.push('250-STARTTLS')
+            response.append('250-STARTTLS')
         if hasattr(self, 'ehlo_hook'):
             warn('Use handler.handle_EHLO() instead of .ehlo_hook()',
                  DeprecationWarning)
             await self.ehlo_hook()
         status = await self._call_handler_hook('EHLO', hostname)
```

Comment:
I think that what I would like to see is an update to this call, maybe something like `ehlo_defaults` or something (actually, given our discussion on #155, I think it would make perfect sense to rename `response` to `ehlo_defaults` and then just pass it to the hook).

Technically I think that's changing up the API and probably needs a doc update as well, but that feels like the right thing to do for this case. Discussion is very welcome, though!

Reply:
+1 on passing the list of responses to EHLO handler

Reply:
Resolved by the latest set of commits.